cscs-checks/microbenchmarks/cpu/alloc_speed/alloc_speed.py

```diff
@@ -7,30 +7,44 @@
 import reframe.utility.sanity as sn
 
 
-@rfm.parameterized_test(['no'], ['2M'])
+@rfm.simple_test
 class AllocSpeedTest(rfm.RegressionTest):
-    def __init__(self, hugepages):
-        self.descr = 'Time to allocate 4096 MB using %s hugepages' % hugepages
-        self.sourcepath = 'alloc_speed.cpp'
-        self.build_system = 'SingleSource'
-        self.build_system.cxxflags = ['-O3', '-std=c++11']
-        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
-                              'eiger:mc', 'pilatus:mc']
-        self.valid_prog_environs = ['PrgEnv-gnu']
-        if hugepages == 'no':
+    hugepages = parameter(['no', '2M'])
+    sourcepath = 'alloc_speed.cpp'
+    valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
+                     'eiger:mc', 'pilatus:mc']
+    valid_prog_environs = ['PrgEnv-gnu']
+    build_system = 'SingleSource'
+    maintainers = ['AK', 'VH']
+    tags = {'production', 'craype'}
+
+    @run_after('init')
+    def set_descr(self):
+        self.descr = (f'Time to allocate 4096 MB using {self.hugepages} '
+                      f'hugepages')
+
+    @run_after('init')
+    def add_valid_systems(self):
+        if self.hugepages == 'no':
             self.valid_systems += ['arolla:cn', 'arolla:pn',
                                    'tsa:cn', 'tsa:pn']
-        else:
-            if self.current_system.name in {'dom', 'daint',
-                                            'eiger', 'pilatus'}:
-                self.modules = ['craype-hugepages%s' % hugepages]
 
-        self.sanity_patterns = sn.assert_found('4096 MB', self.stdout)
-        self.perf_patterns = {
-            'time': sn.extractsingle(r'4096 MB, allocation time (?P<time>\S+)',
-                                     self.stdout, 'time', float)
-        }
-        self.sys_reference = {
+    @run_after('setup')
```

Comment:
This will show you the modules when listing with `-L`.

```suggestion
    @run_after('init')
```